File: SPOJ/LASTDIG/last_digit.c
#include<stdio.h>
#include<stdlib.h>

#define MODULUS 10

static int compute_last_digit(int, int);

int main(void) {
	int test;
	scanf("%d", &test);
	while(test--) {
		int a, b;
		scanf("%d%d", &a, &b);
		printf("%d\n", compute_last_digit(a, b));
	}
	return EXIT_SUCCESS;
}

static int compute_last_digit(int base, int exponent) {
	int result;
	if(!base) {
		result = 0;
	} else if(!exponent) {
		result = 1;
	} else {
		result = 1;
		while(exponent) {
			if(exponent & 1) {
				result = (result * (base)) % MODULUS;
			}
			base = (base * base) % MODULUS;
			exponent >>= 1;
		}
	}
	return result;
}

File: SPOJ/LASTDIG/last_digit.py
"""
Problem Statement: https://www.spoj.com/problems/LASTDIG/
Author: striker
"""

def compute_last_digit(a, b, modulus):
	if a == 1:
		return a
	if b == 1:
		return a % modulus
	result = 1
	while b:
		if b & 1:
			result = (result * a) % modulus
		a = (a * a) % modulus
		b >>= 1
	return result

def main():
	for test in range(int(input().strip())):
		a, b = tuple(map(int, input().strip().split()))
		print(f"{compute_last_digit(a, b, 10)}")

if __name__ == "__main__":
	main()